### PR TITLE
Add ability to get screenshot without writing on disk

### PR DESCRIPTION
**Pull request by MoLLIa**
Sometimes writing screenshot no disk is unnecessary.
Also i've slightly updated tests and docs.

And one more fix for windowHandlePosition test.
I've wrote about it here https://github.com/webdriverio/webdriverio/pull/341#issuecomment-62886775 but looks like you've missed my comment.

Please let me know whether something is wrong.

### CODE REVIEW

File: lib/commands/saveScreenshot.js [hunk 1]
```diff
@@ -31,26 +36,28 @@ module.exports = function saveScreenshot(fileName) {
...
+
+            if (typeof response === 'string') {
+                fs.writeFile(response, screenshot, cb);
+            } else {
+                cb();
+            }
         }
     ], function(err) {
-
-        callback(err, null, response);
```
Comment (christian-bromann) [inline lib/commands/saveScreenshot.js]: we should stay consistent here, meaning that callback should return with error, result and response as parameters here

File: test/spec/desktop/saveScreenshot.js [hunk 2]
```diff
@@ -5,23 +5,30 @@ describe('saveScreenshot', function() {
...
 
         var screenshotPath = path.join(__dirname, '..', '..', '..', 'test.png');
 
         this.client
-            .saveScreenshot(screenshotPath, true)
+            .saveScreenshot(screenshotPath)
             .call(function() {
-
                 fs.exists(screenshotPath, function(fileExists) {
-                    fileExists.should.be.true;
```
Comment (christian-bromann) [inline test/spec/desktop/saveScreenshot.js]: why this change?

File: test/spec/desktop/saveScreenshot.js [hunk 3]
```diff
@@ -5,23 +5,30 @@ describe('saveScreenshot', function() {
...
     before(h.setup());
 
-    var imageSize = {};
-
     it('should take a screenshot and output it on a desired location', function(done) {
 
         var screenshotPath = path.join(__dirname, '..', '..', '..', 'test.png');
 
         this.client
-            .saveScreenshot(screenshotPath, true)
```
Comment (christian-bromann) [inline test/spec/desktop/saveScreenshot.js]: :+1: good catch, was an obsolete parameter

**Comment by christian-bromann** [comment]
@MoLLIa first of all thanks for your contribution. I found couple of things. I still gonna merge this one and will fix them on my own

:+1:

**Comment by MoLLIa** [comment]
You are welcome. Thank you for quick response upon pull request. I am glad to help you.
